parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -549,12 +549,16 @@ ensureGPUDrivers() {
         # no GPU on ARM64
         return
     fi
-
+    
     if [[ "${CONFIG_GPU_DRIVER_IF_NEEDED}" = true ]]; then
-        logs_to_events "AKS.CSE.ensureGPUDrivers.configGPUDrivers" configGPUDrivers
-    else
-        logs_to_events "AKS.CSE.ensureGPUDrivers.validateGPUDrivers" validateGPUDrivers
+        existing_version="$(nvidia-smi | grep "Driver Version" | cut -d' ' -f3)"
+        expects_grid="$(grep -i grid <<< $NVIDIA_DRIVER_IMAGE_TAG)"
+        if [[ ( "*$existing_version*" != "$NVIDIA_DRIVER_IMAGE_TAG" ) || ( -f /usr/bin/nvidia-gridd && -z $expects_grid ) || ( ! -f /usr/bin/nvidia-gridd && -n $expects_grid ) ]]; then
```

Comment:
Having a hard time testing grid drivers. Seems like presence of nvidia-gridd indicates it's grid but running into some issues creating a node with grid drivers to verify this.